### PR TITLE
Add support for support account_recovery_setting

### PR DESCRIPTION
**Pull request by lgallard**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -174,6 +174,21 @@ resource "aws_cognito_user_pool" "pool" {
     }
   }
 
+  # account_recovery_setting
+  dynamic "account_recovery_setting" {
+    for_each = lenght(var.recovery_mechanisms) == 0 ? [] : [1]
```
Comment (hyei) [inline main.tf:179]: Should be length, not lenght

File: README.md [hunk 2]
```diff
@@ -90,13 +90,31 @@ module "aws_cognito_user_pool_complete" {
     }
   ]
 
+  recovery_mechanism = [
```
Comment (hyei) [inline README.md]: Should be recovery_mechanisms, not recovery_mechanism, based on the code

**Review by hyei — CHANGES_REQUESTED** [review]
When I add the ` recovery_mechanisms = []` to my module, it does not get rid of Terraform plan showing that it needs to remove an empty account_recovery_setting block, but not clear if this is something that can be resolved in this module.

I was able to verify adding the following block did explicitly set e-mail only in Cognito, and subsequent terraform plans did not detect any more changes. 

`recovery_mechanisms = [
    {
      name     = "verified_email"
      priority = 1
    },
  ]`

File: main.tf (hunk 1 above)
Comment (lgallard) [reply main.tf]: Thanks!  I just write the code, I couldn't test it (maybe in the future I can include an validate stage in Action)

File: README.md (hunk 2 above)
Comment (lgallard) [reply README.md]: Fixed!

**Comment by lgallard** [comment]
@hyei If it's okay for you, I will merge and ship it!

**Comment by hyei** [comment]
Yes, this solves my issue, thanks!

**Comment by lgallard** [comment]
This fixes #17 and #32

**Pull request merged**